### PR TITLE
Add `XRPLNetwork` property to `XRPClient`

### PR DESCRIPTION
**Pull request by keefertaylor**
## High Level Overview of Change

Add a network parameter in `XRPClient`.

### Context of Change

A future PR will use this network parameter in a Xpring Client which composes XRP and PayID.

This PR also refactors `XRPLNetwork` from the Pay ID package to a common package, since it is now used between XRP and PayID.

Note this is a breaking change. We're already going to do a major semver release to drop the legacy protocol buffer compatibility so we're already going to break all clients of this constructor so we can rip the bandage off all at once.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

No change - client's must update constructor.

## Test Plan

CI

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - `getTransactionStatus` is removed. Please use `getPaymentStatus` instead.
 
 #### Changed
+- `XRPClient` requires a new parameter in it's constructor that identifies the network it is attached to.
```
Comment (amiecorso) [inline CHANGELOG.md:16]: its not it's

File: Tests/PayID/PayIDIntegrationTests.swift [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  /// The PayID to resolve.
+  public static let testPointer = "$dev.payid.xpring.money/hbergren"
+}
+
+/// Integration tests run against a live PayID service.
+final class PayIDIntegrationTests: XCTestCase {
+  func testResolvePaymentPointerKnownPointerMainnet() {
+    let expectation = XCTestExpectation(description: "resolveToXRP completion called.")
+
+    // GIVEN a Pay ID that will resolve on mainnet and a PayID client.
```
Comment (amiecorso) [inline Tests/PayID/PayIDIntegrationTests.swift:14]: Mainnet v.s. mainnet?

File: Tests/PayID/PayIDIntegrationTests.swift [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      expectation.fulfill()
+    }
+
+    self.wait(for: [ expectation ], timeout: 10)
+  }
+
+  func testResolvePaymentPointerKnownPointerTestnet() {
+    let expectation = XCTestExpectation(description: "resolveToXRP completion called.")
+
+    // GIVEN a Pay ID that will resolve on testnet and a PayID client.
```
Comment (amiecorso) [inline Tests/PayID/PayIDIntegrationTests.swift:36]: Testnet vs. testnet?

File: XpringKit/Common/XRPLNetwork.swift [hunk 4]
```diff
@@ -0,0 +1,6 @@
+/// Possible networks to resolve.
+public enum XRPLNetwork: String {
+  case dev = "devnet"
```
Comment (hbergren) [inline XpringKit/Common/XRPLNetwork.swift:3]: Is it common to have lowercase enums in Swift? I know TypeScript favors uppercase enum members.

File: XpringKit/PayID/Generated/APIHelper.swift [hunk 5]
```diff
@@ -0,0 +1,63 @@
+// APIHelper.swift
+//
+// Generated by swagger-codegen
+// https://github.com/swagger-api/swagger-codegen
+//
+
+import Foundation
```
Comment (hbergren) [inline XpringKit/PayID/Generated/APIHelper.swift:7]: Do we need to check in generated code?

File: Cartfile [hunk 6]
```diff
@@ -1,3 +1,5 @@
+github "Alamofire/Alamofire" ~> 4.9.0
```
Comment (hbergren) [inline Cartfile:1]: What is Alamofire?

File: Tests/PayID/PayIDIntegrationTests.swift [hunk 7]
```diff
@@ -0,0 +1,79 @@
+import XCTest
+import XpringKit
+
+extension PaymentPointer {
+  /// The PayID to resolve.
+  public static let testPointer = "$dev.payid.xpring.money/hbergren"
```
Comment (hbergren) [inline Tests/PayID/PayIDIntegrationTests.swift:6]: I assume we change payment pointer / PayID (what do you even call this thing now)? formats later?

Comment (hbergren) [reply Tests/PayID/PayIDIntegrationTests.swift:6]: Or I guess this is blocked on the `payid` repo switching formats, correct?

File: XpringKit/PayID/PayIDClient.swift [hunk 8]
```diff
@@ -0,0 +1,80 @@
+import Alamofire
+import Foundation
+
+/// Implements interaction with a PayID service.
+/// - Warning:  This class is experimental and should not be used in production applications.
+public class PayIDClient {
+  /// The network this PayID client resolves on.
+  private let network: XRPLNetwork
```
Comment (hbergren) [inline XpringKit/PayID/PayIDClient.swift:8]: So this is the non-payment-network agnostic implementation, right?

File: XpringKit/PayID/PayIDClient.swift [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+    let requestBuilder: RequestBuilder<PaymentInformation>.Type = SwaggerClientAPI.requestBuilderFactory.getBuilder()
+    let request = requestBuilder.init(
+      method: "GET",
+      URLString: (url?.string ?? URLString),
+      parameters: parameters, isBody: false
+    ).addHeader(
+      name: "Accept",
+      value: acceptHeaderValue
+    )
```
Comment (hbergren) [inline XpringKit/PayID/PayIDClient.swift:56]: My eyes are glazing over in this file... but I trust you to have it tested 😄

File: pay-id-api-spec/pay-id.v1.yml [hunk 10]
```diff
@@ -0,0 +1,333 @@
...
+          type: string
+        transactionConfirmation:
+          type: string
+      required:
+        - invoiceHash
+        - transactionConfirmation
+openapi: '3.0.0'
+servers:
+  - url: 'https:/'
+    description: ''
```
Comment (hbergren) [inline pay-id-api-spec/pay-id.v1.yml:333]: Does this mean that this `PayID` swaggger file is duplicated in `XpringKit`, `Xpring-JS`, and `Xpring4J`?

File: Tests/PayID/PayIDIntegrationTests.swift (hunk 7 above)
Comment (dino-rodriguez) [reply Tests/PayID/PayIDIntegrationTests.swift:6]: Now that the repo is updated, this can be switched to the new format.

File: CHANGELOG.md (hunk 1 above)
Comment (keefertaylor) [reply CHANGELOG.md:16]: thank you

File: XpringKit/Common/XRPLNetwork.swift (hunk 4 above)
Comment (keefertaylor) [reply XpringKit/Common/XRPLNetwork.swift]: Yes, this is recommended.

Comment (keefertaylor) [reply XpringKit/Common/XRPLNetwork.swift]: https://docs.swift.org/swift-book/LanguageGuide/Enumerations.html

File: XpringKit/PayID/Generated/APIHelper.swift (hunk 5 above)
Comment (keefertaylor) [reply XpringKit/PayID/Generated/APIHelper.swift]: This is a requirement of Carthage, unfortunately. The build system checks out a Github project and builds it. 

If we generate on the fly, then users of this library need all the generation tooling (swagger, protobuf, node, etc) on their local machine, which isn't always true. The errors for this aren't always intuitive and we had a few users early on get stuck.

File: Cartfile (hunk 6 above)
Comment (keefertaylor) [reply Cartfile]: AlamoFire is a popular networking library for Swift: https://github.com/Alamofire/Alamofire

File: Tests/PayID/PayIDIntegrationTests.swift (hunk 7 above)
Comment (keefertaylor) [reply Tests/PayID/PayIDIntegrationTests.swift]: Updated

File: XpringKit/PayID/PayIDClient.swift (hunk 8 above)
Comment (keefertaylor) [reply XpringKit/PayID/PayIDClient.swift]: For now yes.

File: XpringKit/PayID/PayIDClient.swift (hunk 9 above)
Comment (keefertaylor) [reply XpringKit/PayID/PayIDClient.swift]: yup.

File: pay-id-api-spec/pay-id.v1.yml (hunk 10 above)
Comment (keefertaylor) [reply pay-id-api-spec/pay-id.v1.yml]: It does for now. Eventually, I'd like this to live in a single place but I haven't gotten that chance yet.

**Comment by keefertaylor** [comment]
A lot of comments / questions on this are now out-dated but I responded anyway. PTAL.

**Comment by tedkalaw** [comment]
can we close this? iirc we already have this

**Pull request merged**